src/planner/binder/expression/bind_aggregate_expression.cpp

```diff
@@ -225,6 +226,14 @@ BindResult BaseSelectBinder::BindAggregate(FunctionExpression &aggr, AggregateFu
 		}
 	}
 
+	// PushCollation for arg_min/arg_max's children[1] to use collations.
+	if (func.name == "arg_min" || func.name == "arg_max") {
+		D_ASSERT(2 == children.size());
+		auto result_type =
```

Comment:
This needs to be handled in the bind call of the aggregate

Reply:
Thanks but that's not what I meant - this is still hardcoding logic based on function names. This logic needs to be moved to the `Min/Max` and `ArgMin/ArgMax` functions and handled by them specifically in the `bind` call. It should not be in the `FunctionBinder`.

Reply:
Thanks for your reminder! I didn't notice the function has Bind() function before.